src/Symfony/Component/EventDispatcher/ImmutableEventDispatcher.php

```diff
@@ -22,15 +22,26 @@ class ImmutableEventDispatcher implements EventDispatcherInterface
 
     public function __construct(EventDispatcherInterface $dispatcher)
     {
-        $this->dispatcher = $dispatcher;
+        $this->dispatcher = LegacyEventDispatcherProxy::decorate($dispatcher);
     }
 
     /**
      * {@inheritdoc}
+     *
+     * @param string|null $eventName
      */
-    public function dispatch($eventName, Event $event = null)
+    public function dispatch($event/*, string $eventName = null*/)
     {
-        return $this->dispatcher->dispatch($eventName, $event);
+        $eventName = 1 < \func_num_args() ? \func_get_arg(1) : null;
+
+        if (\is_scalar($event)) {
```

Comment:
`dispatch(...func_get_args())` for a while instead?

Reply:
What do you mean? The decorated dispatcher is wrapped by `LegacyEventDispatcherProxy` si we're sure about the way it behaves.

Reply:
never mind, i thought the original dispatcher was always wrapped one